src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptor.java

```diff
@@ -546,4 +547,39 @@ private static String printArgumentTypes(Object[] args) {
         }
     }
 
+    private static boolean permitsFieldGet(@NonNull Whitelist whitelist, @NonNull Field field, @NonNull Object receiver) {
+        if (Modifier.isStatic(field.getModifiers())) {
+            return whitelist.permitsStaticFieldGet(field);
+        }
+        return whitelist.permitsFieldGet(field, receiver);
+    }
```

Comment:
We could put this methods in `Whitelist` itself, but then we would need to come up with a different name.

In retrospect I think we should not have bothered with an API-level distinction between static and instance fields and methods in `Whitelist` and just made implementations check modifiers themselves.

Reply:
I guess so. A bit tricky to change now though, since you would need to use `Util.isOverridden` to accommodate the fact that there are both implementors and (apparently) callers of `Whitelist` in other plugins.

Reply:
Yeah, I do not really think that it is worth making API changes at this point.